Tests/ClientTests/TabLocationViewTests.swift

```diff
@@ -0,0 +1,14 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+
+@testable import Client
+
+final class TabLocationViewTests: XCTestCase {
+    func testTabLocationView_hasNoLeaks() {
+        let tabLocationView = TabLocationView()
```

Comment:
We should set the tabLocationView delegate, before tracking for memory leaks. So not missing much here. You can create a URLBarView to be the TabLocationViewDelegate delegate for our test purpose

Reply:
Do you think this looks, OK bro?

```swift
class TabLocationViewTests: XCTestCase {
    func testDelegateMemoryLeak() {
        let tabLocationView = TabLocationView()
        let delegate = MockTabLocationViewDelegate()
        tabLocationView.delegate = delegate
        delegate.cleanup()
        trackForMemoryLeaks(delegate)
    }
}

// A mock delegate
class MockTabLocationViewDelegate: TabLocationViewDelegate {
    weak var delegate: TabLocationViewDelegate?

    func tabLocationViewDidTapLocation(_ tabLocationView: TabLocationView) {}
    func tabLocationViewDidLongPressLocation(_ tabLocationView: TabLocationView) {}
    func tabLocationViewDidTapReaderMode(_ tabLocationView: TabLocationView) {}
    func tabLocationViewDidTapReload(_ tabLocationView: TabLocationView) {}
    func tabLocationViewDidTapShield(_ tabLocationView: TabLocationView) {}
    func tabLocationViewDidBeginDragInteraction(_ tabLocationView: TabLocationView) {}
    func tabLocationViewDidTapShare(_ tabLocationView: TabLocationView, button: UIButton) {}
    func tabLocationViewDidLongPressReaderMode(_ tabLocationView: TabLocationView) -> Bool {
        return false
    }
    func tabLocationViewDidLongPressReload(_ tabLocationView: TabLocationView) {}
    func tabLocationViewLocationAccessibilityActions(_ tabLocationView: TabLocationView) -> [UIAccessibilityCustomAction]? {
        return nil
    }

    // Cleanup method
    func cleanup() {
        delegate = nil
    }
}

```

Reply:
Yeah bro, thanks. Just made a few comments but this looks good 👍 Almost ready to go